src/rust/dbv2/migrations/2023-11-28-000000_all_balances/up.sql

```diff
@@ -0,0 +1,26 @@
+CREATE VIEW
+  api.user_balances AS
+SELECT DISTINCT
+  ON ("user", market_id, custodian_id) "user",
+  market_id,
+  custodian_id,
+  base_total,
+  base_available,
+  base_ceiling,
+  quote_total,
+  quote_available,
+  quote_ceiling,
+  "time" AS last_update_time,
+  "txn_version" AS last_update_txn_version
+FROM
+  api.balance_updates
+ORDER BY
+  "user" ASC,
+  market_id ASC,
+  custodian_id ASC,
+  txn_version DESC;
```

Comment:
From offline discussion with @CRBl69

Getting user with most balance updates on testnet:

```sql
with temp as (select count(*) as n_entries, "user" from api.balance_updates group by "user") select * from temp order by n_entries desc limit 1;
 n_entries |                                user                                
-----------+--------------------------------------------------------------------
   1085249 | 0x5bc80990c4105df90cb0f15fc145ee83edacf5446036413aca7f69e5c2ccd241
```

Query against DSS `/user_balances?user=eq.0x5bc80990c4105df90cb0f15fc145ee83edacf5446036413aca7f69e5c2ccd241` is slow

Reply:
Creating index: 

```sql
create index test_1 on balance_updates_by_handle(market_id, custodian_id);
```

Reply:
```sql
create index test_1 on balance_updates_by_handle(handle ASC, txn_version DESC);
```

```sql
CREATE VIEW
  api.user_balances AS
SELECT DISTINCT
  ON (handle) "user",
  market_id,
  custodian_id,
  base_total,
  base_available,
  base_ceiling,
  quote_total,
  quote_available,
  quote_ceiling,
  "time" AS last_update_time,
  "txn_version" AS last_update_txn_version
FROM
  balance_updates_by_handle
  NATURAL JOIN market_account_handles
ORDER BY
  handle ASC,
  txn_version DESC;
```